### ISSUE TITLE
How to Create and Update Offers Using JSON_LISTINGS_FEED

### ISSUE THREAD

**Issue by impromani** [jlevers/selling-partner-api#834]
I am currently using the POST_FLAT_FILE_INVLOADER_DATA feed to submit data. This feed performs both actions—creating a new SKU if it doesn't exist and updating the SKU if it already exists.

I am planning to migrate from the **POST_FLAT_FILE_INVLOADER_DATA** feed to **JSON_LISTINGS_FEED**. However, after reviewing the documentation (linked below), I found that the provided examples only demonstrate how to update existing SKUs:
[Listing Workflow Migration Tutorial](https://developer-docs.amazon.com/sp-api/docs/listing-workflow-migration-tutorial#migrating-flat-file-feeds-workflows).

I attempted to use the JSON_LISTINGS_FEED for this purpose, but it only updates existing SKUs and does not create new ones.

Could you please suggest a way to achieve both actions (create and update) in a single API call, similar to how it works with the previous feed?

Below are the feed attributes I am currently using with POST_FLAT_FILE_INVLOADER_DATA:

- sku
- price
- minimum-seller-allowed-price
- maximum-seller-allowed-price
- quantity
- product-id
- product-id-type
- item-condition
- item-note

Any guidance would be greatly appreciated!

**Comment by jlevers** [comment]
I don't think it's possible to create new listings using the `JSON_LISTINGS_FEED` with only the data that's in the `POST_FLAT_FILE_INVLOADER_DATA` feed. Creating products using the Listings API tends to involve specifying a lot of product-type-specific information. In case I'm wrong, though, you could try adding a message with operation type `UPDATE` to the feed, specify the product ID information in addition to prices, quantity, condition, etc, and see if it works:

```json
{
    "messageId": 1,
    "operationType": "UPDATE",
    "productType": "PRODUCT",
    "sku": "MY_SKU_0001",
    "attributes": {
        "externally_assigned_product_identifier": [
            {
                "type": "upc",
                "value": "your-upc-value"
            }
        ],
    }
}
```

**Comment by impromani** [comment]
Hi @jlevers 

Thank you for your response. I tried using "operationType": "UPDATE", but I received the following error:

```
 "issues":[
      {
         "messageId":8,
         "code":"4000004",
         "severity":"ERROR",
         "message":"Creating products is not supported with the Amazon product type 'PRODUCT'. Either specify a specific Amazon product type, or specify a est requirement that only applies to listings."
      },
```

**Comment by jlevers** [comment]
Yeah, I didn't really expect that to work. You can create products through the `JSON_LISTINGS_FEED` or via the [`putListingsItems`](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#putlistingsitem) endpoint, which will both be formatted the same (you'll just package all the attributes you send in `putListingsItems` into a message formatted like the one in my last comment, but with the correct product type for the product you're trying to create).
